src/server/scripts/Spells/spell_holiday.cpp

```diff
@@ -42,75 +42,77 @@ enum SpellsPicnic
+    PrepareAuraScript(spell_love_is_in_the_air_romantic_picnic);
 
-        class spell_love_is_in_the_air_romantic_picnic_AuraScript : public AuraScript
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo(
         {
-            PrepareAuraScript(spell_love_is_in_the_air_romantic_picnic_AuraScript);
+            SPELL_BASKET_CHECK,
+            SPELL_MEAL_PERIODIC,
+            SPELL_MEAL_EAT_VISUAL,
+            SPELL_DRINK_VISUAL,
+            SPELL_ROMANTIC_PICNIC_ACHIEV
+        });
+    }
 
-            void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
-            {
-                Unit* target = GetTarget();
-                target->SetStandState(UNIT_STAND_STATE_SIT);
-                target->CastSpell(target, SPELL_MEAL_PERIODIC, false);
-            }
+    void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* target = GetTarget();
+        target->SetStandState(UNIT_STAND_STATE_SIT);
+        target->CastSpell(target, SPELL_MEAL_PERIODIC);
+    }
 
-            void OnPeriodic(AuraEffect const* /*aurEff*/)
-            {
-                // Every 5 seconds
-                Unit* target = GetTarget();
-                Unit* caster = GetCaster();
+    void OnPeriodic(AuraEffect const* /*aurEff*/)
+    {
+        // Every 5 seconds
+        Unit* target = GetTarget();
 
-                // If our player is no longer sit, remove all auras
-                if (target->GetStandState() != UNIT_STAND_STATE_SIT)
-                {
-                    target->RemoveAura(SPELL_ROMANTIC_PICNIC_ACHIEV);
-                    target->RemoveAura(GetAura());
-                    return;
-                }
+        // If our player is no longer sit, remove all auras
+        if (target->GetStandState() != UNIT_STAND_STATE_SIT)
+        {
+            target->RemoveAurasDueToSpell(SPELL_ROMANTIC_PICNIC_ACHIEV);
+            target->RemoveAura(GetAura());
+            return;
+        }
+
+        target->CastSpell(target, SPELL_BASKET_CHECK); // unknown use, it targets Romantic Basket
+        target->CastSpell(target, RAND(SPELL_MEAL_EAT_VISUAL, SPELL_DRINK_VISUAL));
+
+        bool foundSomeone = false;
+        // For nearby players, check if they have the same aura. If so, cast Romantic Picnic (45123)
+        // required by achievement and "hearts" visual
+        std::list<Player*> playerList;
+        Trinity::AnyPlayerInObjectRangeCheck checker(target, INTERACTION_DISTANCE*2);
+        Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher(target, playerList, checker);
+        Cell::VisitWorldObjects(target, searcher, INTERACTION_DISTANCE * 2);
+        for (std::list<Player*>::const_iterator itr = playerList.begin(); itr != playerList.end(); ++itr)
+        {
+            if (foundSomeone)
+                continue;
```

Comment:
shouldn't this be "break;" instead of "continue;" ?

Reply:
Yes, it seems more optimal, thx